### PR TITLE
New tx fee estimation

### PR DESCRIPTION
**Pull request by ccorcoveanu**

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
 "use strict";
 
 const bech32 = require('bech32');
+const BigNumber = require('bignumber.js');
 
 class Transaction {
-  constructor(nonce = 0, from = '', to = '', value = '', gasPrice = '', gasLimit = '', data = '', chainID='', version = 0) {
+  constructor({nonce = 0, from = '', to = '', senderUsername = '', receiverUsername = '',
```
Comment (bogdan-rosianu) [inline src/transaction.js:7]: maybe each field on a new line?

File: test/account.test.js [hunk 2]
```diff
@@ -37,4 +37,10 @@ describe('account', function() {
 
     assert.deepEqual(secondAccount, firstAccount, "seccondAccount should be the same as firstAccount")
   });
+
+  it("prints reward address", function() {
+    let address = "erd1ezlzq50pq6w2wrpvchg9vd9szgay85ks8jhx9zjksezlae6j5hcsg66xtu";
+    const decoded = bech32.decode(address, 256);
+    console.log(Buffer.from(bech32.fromWords(decoded.words)).toString('hex'));
```
Comment (bogdan-rosianu) [inline test/account.test.js:44]: assert that the result is equal with an expected hex?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: src/transaction.js (hunk 1 above)
Comment (ccorcoveanu) [reply src/transaction.js:7]: done

File: test/account.test.js (hunk 2 above)
Comment (ccorcoveanu) [reply test/account.test.js:44]: Removed this test, it is useless

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**